core/dbt/compilation.py

```diff
@@ -172,6 +172,15 @@ def add_ephemeral_prefix(self, name: str):
         relation_cls = adapter.Relation
         return relation_cls.add_ephemeral_prefix(name)
 
+    def _get_relation_name(self, node: ParsedNode):
+        relation_name = None
+        if (node.resource_type in NodeType.refable() and
+                node.config.materialized != "ephemeral"):
```

Comment:
```suggestion
                not node.is_ephemeral_model
```
based on similar logic [here](https://github.com/fishtown-analytics/dbt/blob/74eec3bdbee1e5be68c800ac400c0e6eac879e64/core/dbt/adapters/base/impl.py#L313-L314)

Reply:
also:
- could you add a quick inline comment to explain this?
- `refable()` excludes `sources`. they're not our primary objective here, but I see benefit in storing a `relation_name` for them, too. what do you think?

Reply:
Sure! I think that's a good idea. Let me explore how can I include the `relation_name` for sources.